pkg/controller/apimanagerrestore/apimanagerrestore_logic_reconciler.go

```diff
@@ -321,35 +242,24 @@ func (r *APIManagerRestoreLogicReconciler) reconcileRestoreAPIManagerInSharedSec
 		return reconcile.Result{}, nil
 	}
 
-	existing := &batchv1.Job{}
-	if r.cr.APIManagerBackupSharedInSecret() {
-		return reconcile.Result{}, nil
-	}
-
-	err := r.ReconcileResource(existing, desired, reconcilers.CreateOnlyMutator)
-	if err != nil {
-		return reconcile.Result{}, err
-	}
-	if existing.Status.Succeeded != *desired.Spec.Completions {
-		r.Logger().Info("Job has still not finished", "Job Name", desired.Name, "Actively running Pods", existing.Status.Active, "Failed pods", existing.Status.Failed)
-		return reconcile.Result{Requeue: true}, nil
-	}
-
-	secret, err := r.sharedBackupSecret()
-	if err != nil {
-		return reconcile.Result{}, err
-	}
-	if secret == nil {
-		r.Logger().Info("shared APIManager backup secret still does not exist", "secret", r.apiManagerRestore.SecretToShareName())
-		return reconcile.Result{Requeue: true}, nil
+	res, err := r.reconcileJob(desired)
+	if res.Requeue || err != nil {
+		return res, err
 	}
 
-	stepFinished := true
-	r.cr.Status.APIManagerBackupSharedInSecret = &stepFinished
-	err = r.UpdateResourceStatus(r.cr)
-	if err != nil {
-		return reconcile.Result{}, err
-	}
+	// TODO placing this check means that we cannot perform cleanup at the end
```

Comment:
Notice this commented code. Originally, after the CreateAPIManagerSharedSecretJob job was performed and successfully finished, a check to verify that the secret existed was performed, and verified that exists. If it did the state boolean was set, otherwise it wasn't.
Now we can't follow the same logic: The reconcileAPIManagerBackupSharedInSecretCleanup method cleans up the secret as one of the latest tasks of the restore. It could happen that that task was performed and a new reconcile loop event executed, making this code fail. This would always expect the secret to exist but we have situations where we do not want to exist. However this is always executed. What would be the way to go now?